PHP templates-specific sniffs (#81)
inpsyde/php-coding-standards: README.md

-------------

## Templates Rules

InpsydeTemplates ruleset contains custom rules targeted to the PHP templates and views.

> To enable the ruleset only for templates the following configuration could be used:

Reviewer: ```suggestion
The InpsydeTemplates ruleset contains custom rules for PHP templates and views.
To enable the ruleset for templates only, the following configuration could be used:
```

```xml
<rule ref="InpsydeTemplates">
    <include-pattern>*/templates/*</include-pattern>
    <include-pattern>*/views/*</include-pattern>
</rule>
```

The following templates-specific rules are available:

> | Sniff name          | Description                                     | Has Config | Auto-Fixable |
> |:--------------------|:------------------------------------------------|:----------:|:------------:|
> | `TrailingSemicolon` | Remove trailing semicolon before close PHP tag. |            |      ✓       |

Reviewer: ```suggestion
| Sniff name          | Description                                       | Has Config | Auto-Fixable |
|:--------------------|:--------------------------------------------------|:----------:|:------------:|
| `TrailingSemicolon` | Remove trailing semicolon before closing PHP tag. |            |      ✓       |
```

-------------

> ## Templates Rules
> 
> InpsydeTemplates ruleset extends Inpsyde ruleset apart from several rules that doesn't make sense in
> the templating context. Several template-specific sniffs are added.
> 
> The recommended way of using InpsydeTemplates ruleset:

Reviewer: ```suggestion
## Template Rules

The `InpsydeTemplates` ruleset extends the standard `Inpsyde` ruleset with some template-specific
sniffs while disabling some rules that are not useful in templating context.

The recommended way to use the `InpsydeTemplates` ruleset is as follows:
```

```xml
<ruleset>
    <file>./src/</file>
    <file>./tests</file>
    <file>./templates</file>

>     <file>./block-views</file>

Reviewer 2: ```suggestion
    <file>./views</file>
```

    <rule ref="Inpsyde">
        <exclude-pattern>*/templates/*</exclude-pattern>
        <exclude-pattern>*/views/*</exclude-pattern>
    </rule>

    <rule ref="InpsydeTemplates">
        <include-pattern>*/templates/*</include-pattern>
        <include-pattern>*/views/*</include-pattern>
    </rule>
</ruleset>
```
The following Inpsyde rules are disabled:

> * NoElse

Reviewer: ```suggestion
* `NoElse`
```

> The following Inpsyde rules are disabled:

Reviewer 2: ```suggestion

The following `Inpsyde` rules are disabled:
```
